docs/workflow/blocks/context_block.rst

```diff
@@ -1,28 +1,44 @@
-Context
-=======
+Context & State
+===============
 
-Use the context block to save data into the global context to make it available for all
+Use the context block to save data into the flow context or global state to make it available for all
 subsequent tasks.
 
+There are two different storage domains that can be written to with this block. 
+
+**Context** is passed from block to block. When processing data in parrallel - during a batch or multiplex flow - context will be different in each branch.
+Blocks will only have access to values written by another block in the same branch. 
+
+**state** is shared by all blocks, and by all flows. When writing to or reading from State, the data is available on two different branches. 
+**state.global** is a datastore that persists for a user's entire session, no matter where they are in the site. 
```

Comment:
Can you clarify 'session'? It's not obvious if it's active memory or cookies, localstorage etc.